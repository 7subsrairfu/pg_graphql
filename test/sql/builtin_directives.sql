begin;
    create table book(
        id int primary key,
        is_verified bool
    );

    insert into public.book(id, is_verified)
    values
        (1, true);

    savepoint a;

    -- Should be skipped
    select jsonb_pretty(
        graphql.resolve($$
            {
              bookCollection {
                edges {
                  node {
                    id
                    isVerified @skip( if: true )
                  }
                }
              }
            }
        $$)
    );

    -- Should not be skipped
    select jsonb_pretty(
        graphql.resolve($$
            {
              bookCollection {
                edges {
                  node {
                    id
                    isVerified @skip( if: false )
                  }
                }
              }
            }
        $$)
    );


    select jsonb_pretty(
        graphql.resolve($$
            query XXX($should_skip: Boolean! ){
              bookCollection{
                edges {
                  node {
                    id
                    isVerified @skip(if: $should_skip)
                  }
                }
              }
            }
          $$,
          '{"should_skip": true}'
        )
    );

    select jsonb_pretty(
        graphql.resolve($$
            query XXX($should_skip: Boolean! ){
              bookCollection{
                edges {
                  node {
                    id
                    isVerified @skip(if: $should_skip)
                  }
                }
              }
            }
          $$,
          '{"should_skip": false}'
        )
    );

rollback;
