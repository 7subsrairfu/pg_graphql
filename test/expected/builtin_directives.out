begin;
    create table book(
        id int primary key,
        is_verified bool
    );
    insert into public.book(id, is_verified)
    values
        (1, true);
    savepoint a;
    -- Should be skipped
    select jsonb_pretty(
        graphql.resolve($$
            {
              bookCollection {
                edges {
                  node {
                    id
                    isVerified @skip( if: true )
                  }
                }
              }
            }
        $$)
    );
          jsonb_pretty           
---------------------------------
 {                              +
     "data": {                  +
         "bookCollection": {    +
             "edges": [         +
                 {              +
                     "node": {  +
                         "id": 1+
                     }          +
                 }              +
             ]                  +
         }                      +
     }                          +
 }
(1 row)

    -- Should not be skipped
    select jsonb_pretty(
        graphql.resolve($$
            {
              bookCollection {
                edges {
                  node {
                    id
                    isVerified @skip( if: false )
                  }
                }
              }
            }
        $$)
    );
                jsonb_pretty                
--------------------------------------------
 {                                         +
     "data": {                             +
         "bookCollection": {               +
             "edges": [                    +
                 {                         +
                     "node": {             +
                         "id": 1,          +
                         "isVerified": true+
                     }                     +
                 }                         +
             ]                             +
         }                                 +
     }                                     +
 }
(1 row)

    select jsonb_pretty(
        graphql.resolve($$
            query XXX($should_skip: Boolean! ){
              bookCollection{
                edges {
                  node {
                    id
                    isVerified @skip(if: $should_skip)
                  }
                }
              }
            }
          $$,
          '{"should_skip": true}'
        )
    );
          jsonb_pretty           
---------------------------------
 {                              +
     "data": {                  +
         "bookCollection": {    +
             "edges": [         +
                 {              +
                     "node": {  +
                         "id": 1+
                     }          +
                 }              +
             ]                  +
         }                      +
     }                          +
 }
(1 row)

    select jsonb_pretty(
        graphql.resolve($$
            query XXX($should_skip: Boolean! ){
              bookCollection{
                edges {
                  node {
                    id
                    isVerified @skip(if: $should_skip)
                  }
                }
              }
            }
          $$,
          '{"should_skip": false}'
        )
    );
                jsonb_pretty                
--------------------------------------------
 {                                         +
     "data": {                             +
         "bookCollection": {               +
             "edges": [                    +
                 {                         +
                     "node": {             +
                         "id": 1,          +
                         "isVerified": true+
                     }                     +
                 }                         +
             ]                             +
         }                                 +
     }                                     +
 }
(1 row)

rollback;
